ImagePickerSheetController/ImagePickerSheetController/ImagePickerSheetController.swift

```diff
@@ -429,13 +429,8 @@ extension ImagePickerSheetController: UICollectionViewDelegate {
         if let maximumSelection = maximumSelection {
             if selectedAssetIndices.count >= maximumSelection,
                 let previousItemIndex = selectedAssetIndices.first {
-                    let deselectedAsset = selectedAssets[previousItemIndex]
-                    delegate?.controller?(self, willDeselectAsset: deselectedAsset)
```

Comment:
To me, it seems odd to call **de**select delegate functions called `willDeselectAsset` and `didDeselectAsset`in the `UICollectionViewDelegate` **select** function `didSelectItemAt`. They are already called in the `UICollectionViewDelegate` **de**select function `didDeselectItemAt `. 

The crash happens because `didDeselectItemAt` calls `selectedAssetIndices.remove(at: index)` which removes the item from `selectedAssets` causing `selectedAssets[previousItemIndex]` to crash with an index out of bounds exception.